tesseract/tesseract_planning/tesseract_process_managers/src/process_managers/raster_dt_process_manager.cpp

```diff
@@ -0,0 +1,354 @@
+/**
+ * @file raster_dt_process_manager.cpp
+ * @brief Plans raster paths with dual transitions
+ *
+ * @author Levi Armstrong
+ * @date August 28, 2020
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2020, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <tesseract_common/macros.h>
+TESSERACT_COMMON_IGNORE_WARNINGS_PUSH
+#include <functional>
+#include <taskflow/taskflow.hpp>
+#include <fstream>
+TESSERACT_COMMON_IGNORE_WARNINGS_POP
+
+#include <tesseract_process_managers/process_managers/raster_dt_process_manager.h>
+#include <tesseract_command_language/instruction_type.h>
+#include <tesseract_command_language/composite_instruction.h>
+#include <tesseract_command_language/plan_instruction.h>
+#include <tesseract_command_language/utils/get_instruction_utils.h>
+
+using namespace tesseract_planning;
+
+struct BasicObserver : public tf::ObserverInterface
+{
+  BasicObserver(const std::string& name) { std::cout << "Constructing observer " << name << '\n'; }
+
+  void set_up(size_t num_workers) override final
+  {
+    std::cout << "Setting up observer with " << num_workers << " workers\n";
+  }
+
+  void on_entry(size_t w, tf::TaskView tv) override final
+  {
+    std::ostringstream oss;
+    oss << "worker " << w << " ready to run " << tv.name() << '\n';
+    std::cout << oss.str();
+  }
+
+  void on_exit(size_t w, tf::TaskView tv) override final
+  {
+    std::ostringstream oss;
+    oss << "worker " << w << " finished running " << tv.name() << '\n';
+    std::cout << oss.str();
```

Comment:
Is this supposed to be here?

Reply:
I left it in one just to keep it around. What was the goal behind using the observer?

Reply:
I was just playing with it. Long term it might be nice to use it for monitoring

Reply:
I will leave it commented out in one just so it stays around until we have time to fully implement it.